rfcs/0052-OpenCLML-integratio-as-BYOC.md

```diff
@@ -0,0 +1,90 @@
+- Feature Name: OpenCL ML integration as BYOC
+- Start Date: 2022-01-13
+- RFC PR: [apache/tvm-rfcs#52](https://github.com/apache/tvm-rfcs/pull/52)
+- GitHub Issue: TBD
+
+
+# Summary
+[summary]: #summary
+
+OpenCL ML is an extension (cl_qcom_ml_ops) over OpenCL spec developed by Qualcomm to accelerate the machine learning at operation level. OpenCL SDK is publicly available at OpenCL Machine Learning Acceleration on Adreno GPU - Qualcomm Developer Network. OpenCL ML leverages deep knowledge of Adreno GPU for significant performance benefits. It offers C based DNN API with compatibility to most of the standard frameworks. Its standard OpenCL features like command queues, buffers, events and supports FP16 and FP32 data types. CLML API calls can be interleaved with other OpenCL kernels (i.e., TVM generated kernels) and dispatched to the same command queue. This extension is compatible with existing OpenCL extensions for importing memory, controlling performance and data access.
```

Comment:
`This extension is compatible with existing OpenCL extensions for importing memory`. Do you mean `cl_qcom_ion_host_ptr` for allocating on the ION memory or what else?

Reply:
CLML tensor object creation happens by creating a ```cl_ml_tensor_desc_qcom``` and then a backing buffer. The backing buffer in general would be a cl_mem object created by clBuffer. Here we can also use ```cl_qcom_ion_host_ptr```.